app/controllers/smart_proxies_controller.rb

```diff
@@ -67,11 +63,18 @@ def destroy
 
   private
 
+  def requested_data(data_name)
+    data = @smart_proxy.public_send(data_name)
```

Comment:
why do we need to go via the proxy at all? can't we just go directly to the status class(es)?